.github/julia/runtests.jl

```diff
@@ -18,11 +18,12 @@ import Uno_jll
 Create a new `AmplNLWriter.Optimizer` object that uses Uno as the backing
 solver.
 """
-function Optimizer()
-    options = String["logger=INFO"]
+function Optimizer(options = String["logger=INFO"])
     return AmplNLWriter.Optimizer(Uno_jll.amplexe, options)
 end
 
+Optimizer_LP() = Optimizer(["logger=INFO", "preset=filtersqp", "subproblem=LP"])
```

Comment:
@cvanaret I get the error `The option with name QP_solver was not found`

Reply:
That's weird. I cannot reproduce the error with the command line. Is it possible that AmplNLWriter passes the options in a different order?

Reply:
> Is it possible that AmplNLWriter passes the options in a different order?

Yes. We store options in a `Dict`, which has an arbitrary iteration order:
https://github.com/jump-dev/AmplNLWriter.jl/blob/1141925e427915be13f45b8b53e3c6415f782e8a/src/AmplNLWriter.jl#L104

Note that JuMP does not have the concept of "ordered" options. I'd classify this as a bug in Uno. 

At minimum, users having to know that the behaviour of Uno depends on the options _and the order in which they appear_ is a massive footgun that is going to cause issues down the track.

Reply:
Ah. In the case of Uno, that's annoying. Precedence lifts any ambiguity in situations like `preset=ipopt option_file=uno.options` and is very useful to overwrite a few options from a preset like we're trying to do here. I think it's very expressive.
What would be your suggestion to address this?

Reply:
In the meantime, you can run the tests with the following options:
```julia
Optimizer_LP() = Optimizer(["LP_solver=HiGHS", "TR_min_radius=1e-8", "TR_radius=10",
"constraint_relaxation_strategy=feasibility_restoration", "filter_type=standard", "globalization_mechanism=TR",
"globalization_strategy=fletcher_filter_method", "l1_constraint_violation_coefficient=1.", "logger=INFO",
"loose_tolerance=1e-6", "progress_norm=L1", "protect_actual_reduction_against_roundoff=no", "residual_norm=L2",
"sparse_format=CSC", "subproblem=LP", "switch_to_optimality_requires_linearized_feasibility=yes",
"tolerance=1e-6"])
```

Reply:
cc @blegat thoughts?

> What would be your suggestion to address this?

You define `preset` to be processed first. Then any other options override. I don't see the ambiguity with `preset` and `option_file`.

You also don't check option validity (does `QP_solver` exist, etc) until all options have been processed.

Reply:
I don't think we have an example of any other solver which has behaviour that depends on the order of the options.

Reply:
We have [SDPA](https://github.com/jump-dev/SDPA.jl/) which has something similar. There are three Modes that give good set of values for parameters but you can also individually set all parameters.
My understanding is that SDPA first process the modes and then set the other parameters. It seems like a good approach that could also be used for the `preset`. If you specifically set a parameter, it seems intuitive that it will take precendence over a `preset`. So the order is done on what's more specific has priority and not the last parameter set has priority.

Reply:
@odow @blegat 
I understand and I realize that this behavior is atypical for a solver. This comes from the fact that there are three possibilities to set options: preset, (partial) option file and individual option.

If a particular option (say `LP_solver`) were to appear in all three possibilities simultaneously, we'd need a way to establish precedence to figure out its actual value. I just figured that the command-line order is more intuitive and predictable that having an implicit order that users have to know about and comply with.

That said, if you think this is an absolute dealbreaker, I'll make the change ;)

Reply:
The configuration file indeed makes it a bit more tricky. If you have an option set both in the config file and explicitly, it's less clear what to do. One option is to error in that case. Otherwise, you could say that the preset is set first, then the option file and then the individual options. If the user sets an option explicitly, my intuition is that it takes precedence over file and preset, it might be confusing if this depends on the order of the arguments.

Reply:
Parse it in this order

1. Option file
2. cmd options
   a. preset
   b. all others 

> That said, if you think this is an absolute dealbreaker, I'll make the change ;)

This is an absolute deal breaker. The solver behaviour should not depend on the order in which options are passed.

Reply:
Fixed in https://github.com/cvanaret/Uno/pull/85.